spec/ruby/core/class/name_spec.rb

```diff
@@ -0,0 +1,12 @@
+require_relative '../../spec_helper'
+require_relative 'fixtures/classes'
+
+describe "Class#name" do
+  it "returns the name as a string for a normal class" do
+    CoreClassSpecs::Record.name.should == "CoreClassSpecs::Record"
+  end
+
+  it "returns nil for a singleton class object" do
+    CoreClassSpecs::Record.singleton_class.name.should be_nil
```

Comment:
Interesting difference compared to `to_s`/`inspect`.
I added a few specs in `spec/ruby/core/module/to_s_spec.rb` recently.